src/gei/Commands/GenerateArchiveCommand.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class GenerateArchiveCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ISourceGithubApiFactory _sourceGithubApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+        private const int _timeoutInHours = 10;
+        private const int _delayInMS = 10000; // 10 seconds
+
+        public GenerateArchiveCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider) : base("generate-archive")
+        {
+            _log = log;
+            _sourceGithubApiFactory = sourceGithubApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
+
+            Description = "Invokes the GitHub Migration API's to generate a migration archive";
+            Description += Environment.NewLine;
+            Description += "Note: Expects GH_PAT and GH_SOURCE_PAT env variables to be set. GH_SOURCE_PAT is optional, if not set GH_PAT will be used instead. This authenticates to the source GHES API.";
+
+            var ghesUrl = new Option<string>("--ghes-url")
+            {
+                IsRequired = false
+            };
+            var githubSourceOrg = new Option<string>("--github-source-org")
+            {
+                IsRequired = true
+            };
+            var sourceRepo = new Option<string>("--source-repo")
+            {
+                IsRequired = true
+            };
+            var noSslVerify = new Option("--no-ssl-verify")
+            {
+                IsRequired = false
+            };
+            var verbose = new Option("--verbose")
+            {
+                IsRequired = false
+            };
+
+            AddOption(ghesUrl);
+            AddOption(githubSourceOrg);
+            AddOption(sourceRepo);
+            AddOption(noSslVerify);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<string, string, string, bool, bool>(Invoke);
+        }
+
+        public async Task Invoke(string ghesUrl, string githubSourceOrg, string sourceRepo, bool noSslVerify = false, bool verbose = false)
+        {
+            _log.Verbose = verbose;
+
+            _log.LogInformation("Generating Migration Archives...");
+            _log.LogInformation($"GHES SOURCE ORG: {githubSourceOrg}");
+
+            if (string.IsNullOrWhiteSpace(ghesUrl))
+            {
+                _log.LogInformation("ghesUrl not provided, defaulting to https://api.github.com");
```

Comment:
Nit: we should standardize this reference to be `--ghes-url`. Just to make it more clear.

Reply:
Sounds good, will add